### PR TITLE
Sumerian scene component

### PR DESCRIPTION
**Pull request by jordanranz**
*Description of changes:*
Amplify XR:
- Sumerian Scene components in React, Angular, Ionic, and Vue
- Updated error handling for unauthorized users in SumerianProvider

Amplify UI:
- XR component styling

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/xr/src/Providers/SumerianProvider.ts [hunk 1]
```diff
@@ -118,7 +118,13 @@ export class SumerianProvider extends AbstractXRProvider {
     } catch (e) {
       logger.debug('No credentials available, the request will be unsigned');
     }
+
+ 
     const apiResponse = await fetch(url, fetchOptions);
+    if (apiResponse.status === 403) {
+      logger.error('Failure to authenticate user');
+      throw(new XRSceneLoadFailure('Failure to authenticate user'));
```
Comment (prestomation) [inline packages/xr/src/Providers/SumerianProvider.ts]: We we include the message throw in the exception from Sumerian? For example, many users could see the message like 'X does not have access to action Y on resource Z' and know right away how to fix, but with this they have to do more digging in the network tools to see the error message, right?

Comment (jordanranz) [reply packages/xr/src/Providers/SumerianProvider.ts]: That is a good call. Let me see if I can pass that message back.

Comment (jordanranz) [reply packages/xr/src/Providers/SumerianProvider.ts]: Updated to display the 403 message in addition to the statement above.

File: packages/aws-amplify-react/src/XR/SumerianScene.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    return new Promise((resolve) => {
+      this.setState(state, resolve)
+    });
+  }
+
+  componentDidMount() {
+    document.addEventListener('fullscreenchange', this.onFullscreenChange.bind(this));
+    document.addEventListener('webkitfullscreenchange', this.onFullscreenChange.bind(this));
+    document.addEventListener('mozfullscreenchange', this.onFullscreenChange.bind(this));
+    document.addEventListener('MSFullscreenChange', this.onFullscreenChange.bind(this));
```
Comment (powerful23) [inline packages/aws-amplify-react/src/XR/SumerianScene.jsx]: Should we remove those listeners when the component is unmounted?

Comment (jordanranz) [reply packages/aws-amplify-react/src/XR/SumerianScene.jsx]: good call out. I will remove these listeners for each framework.

File: packages/aws-amplify-angular/src/components/xr/sumerian-scene-component/sumerian-scene.component.core.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+  async loadAndStartScene() {
+    this.loading = true;
+    const sceneOptions = { 
+      progressCallback: this.progressCallback
+    };
+    await XR.loadScene(this.sceneName, SCENE_DOM_ID, sceneOptions);
+    XR.start(this.sceneName);
+
+    this.loading = false;
```
Comment (haverchuck) [inline packages/aws-amplify-angular/src/components/xr/sumerian-scene-component/sumerian-scene.component.core.ts]: Would this.loading remain true when an error is thrown from XR.loadScene?

File: packages/aws-amplify-vue/src/components/xr/SumerianScene.vue [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+      this.loadPercentage = percentage;
+    },
+    async loadAndStartScene() {
+      this.loading = true;
+      const sceneOptions = { 
+        progressCallback: this.progressCallback
+      };
+      await this.$Amplify.XR.loadScene(this.sceneName, this.SCENE_DOM_ID, sceneOptions);
+      this.$Amplify.XR.start(this.sceneName);
+
```
Comment (haverchuck) [inline packages/aws-amplify-vue/src/components/xr/SumerianScene.vue]: Same question as angular -  is a thrown error from XR.loadScene handled correctly?

Comment (jordanranz) [reply packages/aws-amplify-vue/src/components/xr/SumerianScene.vue]: I have added an error display message for when the scene is unable to load :D Good call.

File: packages/aws-amplify-angular/src/components/xr/sumerian-scene-component/sumerian-scene.component.core.ts (hunk 3 above)
Comment (jordanranz) [reply packages/aws-amplify-angular/src/components/xr/sumerian-scene-component/sumerian-scene.component.core.ts]: yes, I have added an error message to the loading overlay, so this is expected.

**Comment by jordanranz** [comment]
Will wait to merge until docs are approved:
https://github.com/aws-amplify/docs/pull/230

**Pull request merged**